fetch.bs

```diff
@@ -3763,12 +3975,63 @@ steps:
    <var>processBody</var>, <var>processBodyError</var>, and <var>fetchParams</var>'s
    <a for="fetch params">task destination</a>.
   </ol>
+</ol>
 
- <li><p>Wait for either <var>response</var>'s <a for=response>body</a> to be null, or
- <var>response</var>'s <a for=response>body</a>'s <a for=body>stream</a> to be
- <a for=ReadableStream>closed</a> or <a for=ReadableStream>errored</a>, and then set
- <var>request</var>'s <a>done flag</a>.
- <!-- This is really bad and needs to be handled differently at some point. -->
+<p>To <dfn>finalize response</dfn> given a <a for=/>fetch params</a> <var>fetchParams</var> and a
+<a for=/>response</a> <var>response</var>, run these steps:
+
+<ol>
+ <li><p>Let <var>timingInfo</var> be <var>response</var>'s <a for=response>timing info</a>.
+
+ <li><p>If <var>timingInfo</var> is not null, then set <var>timingInfo</var>'s
+ <a for="fetch timing info">response end time</a> to the <a for=/>coarsened shared current time</a>
+ given <var>fetchParams</var>'s <a for="fetch params">cross-origin isolated capability</a>.
+
+ <li><p>Set <var>fetchParams</var>'s <a for="fetch params">request</a>'s
+ <a for=request>done flag</a>.
+
+ <li><p>If <var>fetchParams</var>'s <a for="fetch params">process response done</a> is not null,
+ then <a>queue a fetch task</a> given <var>fetchParams</var>'s
+ <a for="fetch params">process response done</a> and <var>fetchParams</var>'s
+ <a for="fetch params">task destination</a>.
+</ol>
+
+<p>To <dfn export>finalize and report timing</dfn> given a <a for=/>response</a>
+<var>response</var>, a <a for=/>global object</a> <var>global</var>, and a <a for=/>string</a>
+<var>initiatorType</var> (default "<code>other</code>"), run these steps:
+
+<ol>
+ <li><p>If <var>response</var>'s <a for=response>URL list</a> is null or
+ <a for=list lt="is empty">empty</a>, then return.
+
+ <li><p>Let <var>originalURL</var> be <var>response</var>'s <a for=response>URL list</a>[0].
+
+ <li><p>Let <var>timingInfo</var> be <var>response</var>'s <a for=response>timing info</a>.
+
+ <li><p>If <var>timingInfo</var> is null, then return.
+
+ <li><p>Let <var>startTime</var> be <var>timingInfo</var>'s
+ <a for="fetch timing info">fetch start time</a>.
+
+ <li><p>If <var>timingInfo</var>'s <a for="fetch timing info">redirect start time</a> is not 0, then
+ set <var>startTime</var> to <var>timingInfo</var>'s
+ <a for="fetch timing info">redirect start time</a>.
```

Comment:
Sorry I missed this.

Why would we do this for the case where we cannot reveal information?

Reply:
Not sure what you're asking. The `redirect start time` is not revealing, as it occurs before the first redirect and keeps all the redirect info obscure.

Reply:
But doesn't that reveal there was a redirect?

Reply:
No. It sets `startTime` to be the `redirect start time`, so it exposes a `startTime`, and the caller doesn't know that its from a redirect

Reply:
If I do `performance.now()` and then invoke `fetch()` it seems I would be able to tell quite easily it's not the actual start time and instead something from the network.

Reply:
https://github.com/w3c/resource-timing/issues/260 :)
This PR actually fixes that, as `startTime` would mean `redirectStart`, the time before the first redirect, and not `fetchStart`, the time after the last redirect.